bin/mrtrix3.py

```diff
@@ -1,8 +1,51 @@
-import inspect, os, sys, imp
-LIB_FOLDER = os.path.realpath(os.path.join(os.path.dirname(os.path.realpath(inspect.getfile(inspect.currentframe()))), os.pardir, 'lib'))
-fp, pathname, description = imp.find_module("mrtrix3", [ LIB_FOLDER ])
-try:
-  imp.load_module("mrtrix3", fp, pathname, description)
-finally:
-  if fp:
-    fp.close()
+import imp, os, sys
+from distutils.spawn import find_executable
+
+def imported(lib_path):
+  success = False
+  fp = None
+  try:
+    fp, pathname, description = imp.find_module('mrtrix3', [ lib_path ])
+    imp.load_module('mrtrix3', fp, pathname, description)
+    success = True
+  except ImportError:
+    pass
+  finally:
+    if fp:
+      fp.close()
+  return success
+
+# Approach 1:
+# If this file is within the core MRtrix3 installation, it should be easy to navigate to the lib/ directory
+if not imported(os.path.normpath(os.path.join(os.path.dirname(__file__), os.pardir, 'lib'))):
+
+  # Approach 2:
```

Comment:
Why split approaches 1 & 2 when they can both by handled in one go? Approach 2 is what was there before, and it handled approach 1 implicitly...